### PR TITLE
Allow passing in a URL to a tarball with cached git data

### PR DESCRIPTION
**Pull request by sorenh**

### CODE REVIEW

File: build_scripts/make_userdata.sh [hunk 1]
```diff
@@ -59,6 +59,14 @@ if [ -n "${puppet_modules_source_repo}" ]; then
     git merge -m 'Merging Pull Request' test_${pull_request_id}
     popd
   fi
+  if [ -n "${module_git_cache}" ]
```
Comment (bodepd) [inline build_scripts/make_userdata.sh]: a few things here.

* needs to be modules overrides
* need to change librarian puppet from install to update

Comment (bodepd) [inline build_scripts/make_userdata.sh]: * also, it needs to be based on https://github.com/JioCloud/puppet-rjil/pull/435 to be testable

Comment (sorenh) [inline build_scripts/make_userdata.sh]: I don't think it should be modules.overrides. librarian installs in modules.

Comment (sorenh) [inline build_scripts/make_userdata.sh]: Sorry, I totally misread. You're right. I'll fix it.

Comment (bodepd) [inline build_scripts/make_userdata.sh]: also #435 is not ready

**Comment by bodepd** [comment]
will be tested with #437

**Pull request closed without merging**